### PR TITLE
Bump golang.org/x/net from 0.19.0 to 0.23.0

### PR DESCRIPTION
**Pull request by zmerlynn**
Rolls the dependabot PRs into one PR.

### CODE REVIEW

**Comment by markmandel** [comment]
For the examples, can you also bump the version numbers in the `Makefiles`? Then we know new versions will get pushed out as part of the release checklist.

Otherwise, good to merge.

**Comment by zmerlynn** [comment]
> For the examples, can you also bump the version numbers in the `Makefiles`? Then we know new versions will get pushed out as part of the release checklist.
> 
> Otherwise, good to merge.

Done, PTAL

File: examples/simple-game-server/Makefile [hunk 1]
```diff
@@ -38,7 +38,7 @@ WITH_ARM64 ?= 1
 
 mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
 project_path := $(dir $(mkfile_path))
-version := 0.28
```
[the remark this answers is not in the document]
Comment (Kalaiselvi84) [reply examples/simple-game-server/Makefile:41]: I will push it now

[the remark this answers is not in the document]
Comment (zmerlynn) [reply examples/simple-game-server/Makefile:41]: I don't think we need two version bumps. @Kalaiselvi84 can pick up this PR and push, or vice-versa, right?

[the remark this answers is not in the document]
Comment (markmandel) [reply examples/simple-game-server/Makefile:41]: Worth noting - we are not `PullAlways` on our test images (although we could be temporally when iterating on an image in the PR)

This means that nodes will cache images, which means that behaviour will not update with each change.

See: https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy

**Review by markmandel — APPROVED** [review]
SHIP IT!

SHIP IT

**Pull request merged**